crates/core/component/dex/src/lp/position.rs

```diff
@@ -29,6 +29,11 @@ pub struct Position {
     /// duplicate position [`Id`]s, so it can track position ownership with a
     /// sequence of stateful NFTs based on the [`Id`].
     pub nonce: [u8; 32],
+    /// Set to `true` if a position is a limit-order, meaning that it will be closed
+    /// after being filled against.
+    pub close_on_fill: bool,
+    /// Set to the `asset::Id` of the asset provided in a limit-order, `None` otherwise.
+    pub limit_order_asset: Option<asset::Id>,
```

Comment:
@hdevalence I've added this field to our `Position` struct. It's relevant when the LP is a limit-order. It is initialized to `None`, and will be tagged with the `asset::Id` of the inventory that the user provides when opening the position. That means that when we handle a position write, we know which asset to inspect to check their reserves. The drawback of course is that `Position` and `pb::Position` don't match perfectly. Curious to read your thoughts on this, when you have time.

Reply:
Wait, why do we need an extra field, rather than just using the close_on_fill bool, and closing the position when either asset is set to zero?

Reply:
In #2744 we have put_position become async so it can read the current state, that change seems relevant here even if the liquidity indexing part isn't implemented

Reply:
Ah yes, if `put_position` can read the current state then we can pull the position from the index, check the state of its reserves, determine that the asset provided is `1` or `2`, and inspect the reserves in `put_position` to figure out if that corresponds to a total fill.